### ISSUE TITLE
Docker 1.13.0 - segfault?

### ISSUE THREAD

**Issue by hashedvalue** [docker/docker#31568]
Hi All.

I am using:

```
Server Version: 1.13.0
Storage Driver: devicemapper
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Kernel Version: 3.10.0-514.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
Containers: 40
 Running: 39
 Paused: 0
 Stopped: 1
```

And have a segfault several times per day:

```
Mar 03 15:12:40 host.domain.net docker[11397]: fatal error: unexpected signal during runtime execution
Mar 03 15:12:40 host.domain.net docker[11397]: [signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x7f471d387fbe]
Mar 03 15:12:40 host.domain.net docker[11397]: runtime stack:
Mar 03 15:12:40 host.domain.net docker[11397]: runtime.throw(0x192633d, 0x2a)
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/panic.go:566 +0x95
Mar 03 15:12:40 host.domain.net docker[11397]: runtime.sigpanic()
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/sigpanic_unix.go:12 +0x2cc
Mar 03 15:12:40 host.domain.net docker[11397]: goroutine 87227 [syscall, locked to thread]:
Mar 03 15:12:40 host.domain.net docker[11397]: runtime.cgocall(0x14737c0, 0xc422c1ed80, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/cgocall.go:131 +0x110 fp=0xc422c1ed50 sp=0xc422c1ed10
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/daemon/logger/*journald*._Cfunc_free(0x7f4658000c00)
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/daemon/logger/journald/_obj/_cgo_gotypes.go:108 +0x41 fp=0xc422c1ed80 sp=0xc422c1ed50
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/daemon/logger/journald.(*journald).drainJournal(0xc421f1fea0, 0xc4224b8210, 0x0, 0x0, 0x0, 0xffffffffffffffff, 0x1, 0x7f466c0008c0, 0x7f4658000c00, 0x7f465800
0c00)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/logger/journald/read.go:239 +0x263 fp=0xc422c1eeb0 sp=0xc422c1ed80
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/daemon/logger/journald.(*journald).followJournal.func1(0x7f466c0008c0, 0x10400000022, 0xc420b86078, 0xc421f1fea0, 0xc4224b8210, 0x0, 0x0, 0x0, 0xfffffffffffff
fff, 0x1)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/logger/journald/read.go:253 +0x11d fp=0xc422c1ef60 sp=0xc422c1eeb0
Mar 03 15:12:40 host.domain.net docker[11397]: runtime.goexit()
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc422c1ef68 sp=0xc422c1ef60
Mar 03 15:12:40 host.domain.net docker[11397]: created by github.com/docker/docker/daemon/logger/journald.(*journald).followJournal
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/logger/journald/read.go:269 +0x158
Mar 03 15:12:40 host.domain.net docker[11397]: goroutine 1 [chan receive, 4492 minutes]:
Mar 03 15:12:40 host.domain.net docker[11397]: main.(*DaemonCli).start(0xc420486c30, 0x0, 0x18ffda8, 0x17, 0xc42048a000, 0xc42045ccd0, 0xc4200a0b40, 0x0, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/daemon.go:326 +0x183f
Mar 03 15:12:40 host.domain.net docker[11397]: main.runDaemon(0x0, 0x18ffda8, 0x17, 0xc42048a000, 0xc42045ccd0, 0xc4200a0b40, 0x17, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:86 +0xb2
Mar 03 15:12:40 host.domain.net docker[11397]: main.newDaemonCommand.func1(0xc420419200, 0xc420098a80, 0x0, 0x17, 0x0, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:42 +0x71
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).execute(0xc420419200, 0xc420098010, 0x17, 0x17, 0xc420419200, 0xc420098010)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:646 +0x26d
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420419200, 0x16edac0, 0xc42045cc01, 0xc42046eeb0)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:742 +0x377
Mar 03 15:12:40 host.domain.net docker[11397]: github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420419200, 0xc42046eeb0, 0xc420070058)
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:695 +0x2b
Mar 03 15:12:40 host.domain.net docker[11397]: main.main()
Mar 03 15:12:40 host.domain.net docker[11397]: /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:106 +0xe2
Mar 03 15:12:40 host.domain.net docker[11397]: goroutine 17 [syscall, 4492 minutes, locked to thread]:
Mar 03 15:12:40 host.domain.net docker[11397]: runtime.goexit()
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/asm_amd64.s:2086 +0x1
Mar 03 15:12:40 host.domain.net docker[11397]: goroutine 3 [syscall, 3200 minutes]:
Mar 03 15:12:40 host.domain.net docker[11397]: os/signal.signal_recv(0x2426880)
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/runtime/sigqueue.go:116 +0x157
Mar 03 15:12:40 host.domain.net docker[11397]: os/signal.loop()
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/os/signal/signal_unix.go:22 +0x22
Mar 03 15:12:40 host.domain.net docker[11397]: created by os/signal.init.1
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/os/signal/signal_unix.go:28 +0x41
Mar 03 15:12:40 host.domain.net docker[11397]: goroutine 7 [syscall, 4492 minutes]:
Mar 03 15:12:40 host.domain.net docker[11397]: syscall.Syscall6(0xf7, 0x1, 0x2c95, 0xc420076100, 0x1000004, 0x0, 0x0, 0x0, 0x0, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
Mar 03 15:12:40 host.domain.net docker[11397]: os.(*Process).blockUntilWaitable(0xc42051c810, 0x0, 0x0, 0x0)
Mar 03 15:12:40 host.domain.net docker[11397]: /usr/local/go/src/os/wait_waitid.go:28 +0xbc
```

I checked os logs but nothing fancy there. I noted though that out of 39 containers 33 are using journald for logging and:

```
# journalctl -u docker -x -l --since "2017-03-03 00:00:00" --until "2017-03-04 00:00:00"| wc -l
62542
```

Can someone help me with troubleshooting?

**Comment by thaJeztah** [comment]
Can you try updating to the current version? Docker 1.13.1 and 17.03 (which, other than changing to the new version scheme, only contains bug fixes since 1.13.1), contain various bug fixes, also related to logging.

**Comment by amatsus** [comment]
This issue is fixed in v17.03.0-ce. 
- Fix a possible crash when using journald #31231 #31263

**Comment by hashedvalue** [comment]
@amatsus will this issue be fixed also for v1.13? In large corporate infrastructure upgrade of repositories and packages because of the changes in regard to release management which you introduced lately (https://blog.docker.com/2017/03/docker-enterprise-edition/) can take some time.

**Comment by cpuguy83** [comment]
@hasedvalue 17.03 is what 1.13.2 would have been, no new features, only critical patches.

The old repos still contain the docker-engine package with these fixes, just under the new version 17.03.

Closing since this is resolved but feel free to continue the discussion here.
